bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -107,9 +110,21 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
      */
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
-        logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
-            writeToKNX(itemName, command);
+        logger.trace(
+                "1. internalReceiveCommand from openHAB (item='{}', command='{}', operatingEnvironmentIsFelix = '{}'",
+                itemName, command.toString(), operatingEnvironmentIsFelix);
+
+        // openHAB1: After receiving a command from openHAB1, NO update event is triggered later.
+        // openHAB2: After receiving a command from openHAB2, ALWAYS update event is triggered later.
+        // In openHAB2 we ignore this command and wait for receiving the update event from openHAB,
+        // then only this is sent to the KNX bus by openHAB2
+
+        if (getEnvironmentIsFelix() == true) {
+            // ignore internalReceiveCommand in openHAB 2
```

Comment:
I worry that relying on the auto-update logic, and relying on a stack trace to determine if it's running on Felix, are only working around a deeper root cause.  I think there may be a deeper issue in auto-update handling, or possibly in how echoes are detected, and this PR only masks it.